Upgrade to ocean v4 RC 1

mihails-strasuns-sociomantic opened the pull request:
```
* submodules/ocean v4.0.0-alpha.0(6941094)...v4.0.0-alpha.3(186faa8) (110 commits)
  > `NotifyingQueue.push`: Don't use `copy`
  > Add alias to fix subclass overload set
  > Change queue methods to use `void[]` buffers with `const` where applicable
  > Small type fixes
  > Add missing cast in unittest
  (...)
```

There is one change that I don't know how to handle (unregisterSocker override),
please advise.

mihails-strasuns-sociomantic commented:
As for D2 failures - I think we should revert change that made `pop` return `const(void)[]` instead of a `void[]`. It causes lot of transitive const damage and is not necessary because queue internally is known to store mutable data. FYI @david-eckardt-sociomantic

Burgos commented:
I think just calling `this.connection.shutdown` (https://github.com/sociomantic-tsunami/swarm/blob/v4.x.x/src/swarm/neo/connection/ConnectionBase.d#L1005) will be enough. Pretty much, we want to unregister from the epoll anything that could reference this socket.

Burgos commented:
(I would confirm this with @david-eckardt-sociomantic )

gavin-norman-sociomantic commented:
>I think just calling this.connection.shutdown (https://github.com/sociomantic-tsunami/swarm/blob/v4.x.x/src/swarm/neo/connection/ConnectionBase.d#L1005) will be enough. Pretty much, we want to unregister from the epoll anything that could reference this socket.

`connection.shutdown` will actually shut down the socket and the send/receive loops. I'm not 100% sure whether that's really what we want to do here.

Another possibility would be to just do this: https://github.com/sociomantic-tsunami/swarm/blob/v4.x.x/src/swarm/neo/connection/ConnectionBase.d#L1127-L1130

@david-eckardt-sociomantic what do you think?

david-eckardt-sociomantic commented:
`unregisterSocket` should only unregister the socket from epoll in the way @gavin-norman-sociomantic suggested:
```d
if (socket.is_registered)
    epoll.unregister(socket);
```

mihails-strasuns-sociomantic committed:
Upgrade to ocean v4.0.0 RC 1

* submodules/beaver v0.3.0(3f00ea2)...v0.3.2(82f8c8f) (5 commits)
  > Fix spelling of BEAVER_DOCKER_IMG
  > bintray: Improve error message when -d is invalid
  > bintray: Fix typo in error message
  > codecov: Add prefix to all sent flags
  > codecov: Properly sanitize sent flags

* submodules/makd v2.0.2(6278813)...v2.1.2(430cd09) (6 commits)
  > Merge tag v1.11.2 into v2.1.x v1.11.2
  > Merge tag v1.11.1 into v2.1.x v1.11.1
  > Merge tag v2.0.3 into v2.1.x v2.0.3
  > Merge tag v1.11.0 into v2.x.x Features --------
  > Merge tag v2.0.2 into v2.x.x v2.0.2
  (...)

* submodules/ocean v4.0.0-alpha.0(6941094)...v4.0.0-rc.1(403b7e2) (142 commits)
  > Fix enum integral promotion deprecation in PHPSerializer
  > Fix enum integral promotion deprecation
  > Fix deprecation for pointer arithmetics
  > Upgrade tested upstream compiler version
  > Simplify --version and add --build-info
  (...)

mihails-strasuns-sociomantic commented:
Updated.

nemanja-boric-sociomantic commented on src/swarm/client/connection/RequestOverflow.d:
```diff
@@ -293,11 +293,11 @@ public class DiskOverflow : IRequestOverflow
         ***********************************************************************/
 
-        public void restore ( ubyte[] stored );
+        public void restore ( void[] stored );
```
Breaking change? :police_car:

nemanja-boric-sociomantic replied:
Ha, this is the interesting case where the swarm release notes are still relevant for the app users.

mihails-strasuns-sociomantic replied:
Yup, this breaking change is already documented in ocean though (which is what forces it in swarm).

gavin-norman-sociomantic replied:
Isn't the change in ocean just an implementation detail, from the POV of a user of this module in swarm? i.e. there should be a migration note here.

gavin-norman-sociomantic commented:
D2 is unhappy.

mihails-strasuns-sociomantic committed:
Switch to upstream DMD 2.078

mihails-strasuns-sociomantic commented:
That is D-APT package version issue we mentioned, added commit to switch to 2.078 to workaround it
